src/Kmd.Logic.Gateway.Automation/PreValidation/IValidation.cs

```diff
@@ -7,5 +7,7 @@ namespace Kmd.Logic.Gateway.Automation
     public interface IValidation
     {
         Task<bool> ValidateAsync(GatewayDetails gatewayDetails);
+
+        public List<PublishResult> PublishResults { get; }
```

Comment:
I would return publish results from ValidateAsync, instead of the context of the validator. You can create class `GatewayValidationResult` and nest there properties:
* `public bool IsErrors { get; }`
* `public IEnumerable<PublishResult> { get; }`

Trust me, it will be much cleaner ;)

Reply:
Changed as suggested.